sdk/identity/azure-identity/azure/identity/_credentials/user.py

```diff
@@ -37,29 +37,27 @@ class DeviceCodeCredential(PublicClientCredential):
     https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-device-code
 
     :param str client_id: the application's ID
-    :param prompt_callback:
-        (optional) A callback enabling control of how authentication instructions are presented.
-        Must accept arguments (``verification_uri``, ``user_code``, ``expires_in``):
-            - ``verification_uri`` (str) the URL the user must visit
-            - ``user_code`` (str) the code the user must enter there
-            - ``expires_in`` (int) the number of seconds the code will be valid
-        If not provided, the credential will print instructions to stdout.
 
     Keyword arguments
         - **authority**: Authority of an Azure Active Directory endpoint, for example 'login.microsoftonline.com', the
           authority for Azure Public Cloud (which is the default). :class:`~azure.identity.KnownAuthorities` defines
           authorities for other clouds.
-        - **tenant (str)** - tenant ID or a domain associated with a tenant. If not provided, defaults to the
-          'organizations' tenant, which supports only Azure Active Directory work or school accounts.
+        - **tenant_id (str)**: an Azure Active Directory tenant ID. Defaults to the 'organizations' tenant, which can
+          authenticate work or school accounts.
```

Comment:
This should probably mention that it can only be unspecified in the case that the app is multi-tenanted.